APCAppCore/APCAppCore/UI/Model/APCOnboardingManager.m

```diff
@@ -0,0 +1,238 @@
+        //[(APCAppDelegate *)[UIApplication sharedApplication].delegate afterOnBoardProcessIsFinished];     // TODO: Untwine this spaghetti
+        [[NSNotificationCenter defaultCenter] postNotificationName:APCUserSignedInNotification object:self];
+    } else {
+        self.user.signedUp = YES;
+        [[NSNotificationCenter defaultCenter] postNotificationName:APCUserSignedUpNotification object:self];
+    }
+    
+    [[NSNotificationCenter defaultCenter] postNotificationName:APCUserDidConsentNotification object:self];
+}
+
+
+#pragma mark - Permissions
+
+- (APCPermissionsManager *)permissionsManager {
+    if (!_permissionsManager) {
+        _permissionsManager = [self createPermissionsManager];
+    }
+    return _permissionsManager;
+}
+
+- (APCPermissionsManager *)createPermissionsManager {
+    APCPermissionsManager *manager = [APCPermissionsManager new];
+    
+    // set HealthKit characteristic types to read
+    NSMutableArray *types = [NSMutableArray arrayWithCapacity:3];
+    for (NSNumber *type in self.userProfileElements) {
+        switch (type.integerValue) {
+            case kAPCUserInfoItemTypeBiologicalSex:
+                [types addObject:HKCharacteristicTypeIdentifierBiologicalSex];
+                break;
+            case kAPCUserInfoItemTypeDateOfBirth:
+                [types addObject:HKCharacteristicTypeIdentifierDateOfBirth];
+                break;
+            case kAPCUserInfoItemTypeBloodType:
+                [types addObject:HKCharacteristicTypeIdentifierBloodType];
+                break;
+        }
+    }
+    manager.healthKitCharacteristicTypesToRead = types;
+    
+    return manager;
+}
+
+
+#pragma mark - User Profile
+
+- (NSArray *)userProfileElements {
+    if (!_userProfileElements) {
+        _userProfileElements = [self createUserProfileElements];
+    }
+    return _userProfileElements;
+}
+
+- (NSArray *)createUserProfileElements {
+    return @[
+        @(kAPCUserInfoItemTypeEmail),
+        @(kAPCUserInfoItemTypeBiologicalSex),
+        @(kAPCUserInfoItemTypeHeight),
+        @(kAPCUserInfoItemTypeWeight),
+        @(kAPCUserInfoItemTypeWakeUpTime),
+        @(kAPCUserInfoItemTypeSleepTime),
+    ];
+}
+
+
+#pragma mark - APCOnboardingDelegate
+
+- (APCScene *)onboarding:(APCOnboarding * __nonnull)onboarding sceneOfType:(NSString * __nonnull)type {
+    NSParameterAssert(onboarding);
+    NSParameterAssert([type length] > 0);
+    
+    // Sign Up
+    if ([type isEqualToString:kAPCSignUpInclusionCriteriaStepIdentifier]) {
+        if ([_provider respondsToSelector:@selector(inclusionCriteriaSceneForOnboarding:)]) {
+            return [_provider performSelector:@selector(inclusionCriteriaSceneForOnboarding:) withObject:onboarding];
+        }
+        return nil;
+    }
+    if ([type isEqualToString:kAPCSignUpEligibleStepIdentifier]) {
+        return [[APCScene alloc] initWithName:@"APCEligibleViewController" inStoryboard:kAPCOnboardingStoryboardName];
```

Comment:
Ah; the downside of doing this is that you would need to import 9 header files just to be able to call `+class`.

I'd prefer to leave this as strings: becomes important when you subclass and do _not_ want to use certain steps. E.g. we don't have a _sign-in_ step so we don't even include that view controller in the project. Does this make sense?